base_bank_branch/views/res_bank_views.xml

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="view_res_bank_tree_inherit" model="ir.ui.view">
+        <field name="name">res.bank.tree.branch</field>
+        <field name="model">res.bank</field>
+        <field name="inherit_id" ref="base.view_res_bank_tree"/>
+        <field name="arch" type="xml">
+            <field name="country" position="after">
+                <field name="is_branch"/>
+                <field name="parent_id" invisible="1"/>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_res_bank_form_inherit" model="ir.ui.view">
+        <field name="name">res.bank.form.branch</field>
+        <field name="model">res.bank</field>
+        <field name="inherit_id" ref="base.view_res_bank_form"/>
+        <field name="arch" type="xml">
+            <group name="bank_details" position="before">
+                <group
+                    name="branch_option"
+                    col="4"
+                    attrs="{'invisible': [('has_childs', '=', True)]}"
+                >
+                    <field name="is_branch"/>
+                    <field
+                        name="parent_id"
+                        placeholder="Main Bank.."
+                        attrs="{'invisible': [('is_branch', '=', False)], 'required': [('is_branch', '=', True)]}"
+                        domain="[('is_branch', '=', False)]"
```

Comment:
I guess, you should not allow to select active/current bank as main bank, because user won't be able to undo action - fields are hidden https://focusate-misc-feature-bank-branch-ala-2290945.dev.odoo.com/web?debug=1#id=3&action=418&model=res.bank&view_type=form&cids=1&menu_id=4

Reply:
Didn't understand what you mean by that. You mean if you hack main bank to be branch or something?